ynr/apps/moderation_queue/tests/test_changes_for_review.py

```diff
@@ -257,6 +261,142 @@ def test_users_with_past_approved_edits_dont_need_review(self):
+        #     user=self.other_user,
+        # )
+
+        form_data = response.forms["person-details"]
+        form_data["favourite_biscuit"] = "Chocolate Chip"
+        form_data["source"] = "Another biscuit update for testing purposes"
+        form_data.submit()
+        example_person.refresh_from_db()
+
+        self.assertEqual(example_person.favourite_biscuit, "Chocolate Chip")
+
+        form_data = response.forms["person-details"]
+        form_data["favourite_biscuit"] = "Coconut"
+        form_data["source"] = "Another biscuit update for good measure"
+        form_data.submit()
+        example_person.refresh_from_db()
+
+        self.assertEqual(example_person.favourite_biscuit, "Coconut")
+
+        versions_data = example_person.versions
+        self.assertEqual(len(versions_data), 3)
+        self.assertEqual(LoggedAction.objects.count(), 3)
+        self.assertEqual(
+            LoggedAction.objects.filter(action_type="person-update").count(), 3
+        )
+
+        response = self.app.get("/", user=self.user)
+        self.user.is_active = False
+        self.user.save()
+        response = self.app.get("/")
+        self.assertIn("Sign in to edit", response.text)
+
+    def revert_edit_to_person(self, user, example_person):
+        self.user = User.objects.get(id=3)
+
+        # create a user history
+        for i in range(4):
+            la = LoggedAction.objects.create(
+                id=(1500 + i),
+                user=self.user,
+                action_type=ActionType.PERSON_UPDATE,
+                person=example_person,
+                popit_person_new_version=random_person_id(),
+                source="Just for tests...",
+            )
+
+        response = self.app.get("/", user=self.user)
+        self.assertIn("Signed in as", response.text)
+        self.assertEqual(self.user.username, "alice")
+
+        versions_data = example_person.versions
+        self.assertEqual(len(versions_data), 3)
+
+        # we want the revert to come from a different user
+        response = self.app.get(
+            "/person/{person_id}/update".format(person_id=example_person.id),
+            user=self.user,
+        )
+
+        version_id = example_person.versions[1]["version_id"]
+        revert_form = response.forms[
+            "revert-form-{version_id}".format(version_id=version_id)
+        ]
+        revert_form[
+            "source"
+        ] = "Reverting to version {version_id} for testing purposes"
+        revert_form.submit()
+        example_person.refresh_from_db()
+        revert_qs = LoggedAction.objects.filter(
+            action_type="person-revert", user=self.user, person=example_person
+        )
+        self.assertEqual(revert_qs.count(), 1)
+        # la = revert_qs[0]
+        # la.flagged_type = "needs_review_due_to_reverted_edit_by_another_user"
+        # la.flagged_reason = "Revert of an edit by another user"
```

Comment:
@michaeljcollinsuk I'm still missing something - where should the `"person-revert" LoggedAction` actually get a `flagged_type`? It doesn't seem right to add it to the helper method. Maybe I'm missing another method that performs this action?

Reply:
Although I thought it was being set here https://github.com/DemocracyClub/yournextrepresentative/blob/425bb67b71bb9a0fb091c07549a9d98e071e6984/ynr/apps/candidates/models/db.py#L206

Reply:
Have a look at the LoggedAction model - there is a method on [set_review_required](set_review_required) (which is called by the `save` method) which checks if the action being made should be reviewed by called the ReviewType's `need_review` method.

So in your tests if this is not happening correctly, your login in the `needs_review` method might need changing?